src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_utils.cpp

```diff
@@ -131,6 +137,10 @@ bool BrgemmConfig::is_amx() const {
 
 void BrgemmConfig::validate() const {
     OPENVINO_ASSERT(m_isa != isa_undef, "ISA is undefined");
+    OPENVINO_ASSERT(one_of(m_src_dt, element::f32, element::bf16, element::f16, element::u8, element::i8),
+                    "Brgemm doesn't support weights element type: " + m_wei_dt.get_type_name());
+    OPENVINO_ASSERT(one_of(m_wei_dt, element::f32, element::bf16, element::f16, element::i8),
+                    "Brgemm doesn't support weights element type: " + m_wei_dt.get_type_name());
```

Comment:
What do you think: should we also check `orig_wei_dt` here? For example, we don' support f8, nf4 etc... 🤔

Reply:
You are right, it's better to add the corresponding check. Added, thanks!